File: src/frontend/TexEdit.Utils/Browser.cs
/*
 *   Copyright (c) 2024 Jack Bennett.
 *   All Rights Reserved.
 *
 *   See the LICENCE file for more information.
 */

using System.Diagnostics;

namespace TexEdit.Utils {
    public static class Browser {
        public static class Urls {
            public static string GitRepository => "https://github.com/kosude/texedit";
        }

        public static bool OpenWebPage(string url) {
            Process proc = new();

            Logging.Log($"Attempting to open webpage at {url}");

            try {
                proc.StartInfo.UseShellExecute = true;
                proc.StartInfo.FileName = url;
                proc.Start();
            } catch (Exception e) {
                Logging.Error(e.Message);
                return false;
            }

            return true;
        }
    }
}

File: frontend/src/browser.rs
/*
 *   Copyright (c) 2024 Jack Bennett.
 *   All Rights Reserved.
 *
 *   See the LICENCE file for more information.
 */

use std::ffi::OsStr;

use crate::log;

pub fn github_url() -> &'static str {
    "https://github.com/kosude/texedit"
}

pub fn open_url(url: impl AsRef<OsStr>) -> bool {
    match open::that(url) {
        Ok(()) => true,
        Err(e) => {
            log::error(e.to_string());
            return false;
        }
    }
}
